include/zephyr/net/net_if.h

```diff
@@ -576,6 +600,34 @@ static inline bool net_if_flag_is_set(struct net_if *iface,
 	return atomic_test_bit(iface->if_dev->flags, value);
 }
 
+/**
+ * @brief Set an operational state on an interface
+ *
+ * @param iface Pointer to network interface
+ * @param oper_state Operational state to set
+ */
+static inline void net_if_oper_state_set(struct net_if *iface,
+					 enum net_if_oper_state oper_state)
+{
+	NET_ASSERT(iface);
+
+	iface->if_dev->oper_state = oper_state;
+}
```

Comment:
Minor thing, but I would suggest that we check the value so that it is within allowed range.

Reply:
We could also return the new value so that we could catch any errors. So if the returned value != oper_state, then the value was invalid. Perhaps all this checking is far fetched but it would be nice to catch all these errors.

Reply:
Ok, I've updated `net_if_oper_state_set()` and added a simple error checking.